pkg/printers/printer.go

```diff
@@ -1,9 +1,141 @@
 package printers
 
 import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+	"github.com/golangci/golangci-lint/pkg/report"
 	"github.com/golangci/golangci-lint/pkg/result"
 )
 
-type Printer interface {
+const defaultFileMode = 0644
+
+type printer interface {
```

Comment:
`pinter` interface and `Printer` type is confusing but it works so well for both so I don't have any better suggestion 😿 

Maybe not worth it but we could call one of them `issuePrinter`, `resultPrinter`, `reportsPrinter` or similar? And since the interface is private and only internal I think if so that should be the one renamed.

Reply:
I share the same feeling as you, this naming is not simple.
I rename it to `issuePrinter`